src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -48,6 +50,25 @@ internal abstract class SourceParameterSymbol : SourceParameterSymbolBase
                     identifier.Parent.GetLocation());
             }
 
+            if (shouldPlaceIsConstModifier && refKind == RefKind.RefReadOnly && (owner.IsVirtual || owner.IsAbstract))
```

Comment:
>(owner.IsVirtual || owner.IsAbstract) [](start = 80, length = 37)

It feels like this condition should be checked by the caller, otherwise ```shouldPlaceIsConstModifier``` shouldn't be true.